pkg/auth/rbac.go

```diff
@@ -433,23 +437,41 @@ func internalMatchNamespace(srcNs string, namespaces []*security.StringMatch) bo
 	return false
 }
 
-func (r *Rbac) buildConnV4(tupleV4 *bpfSockTupleV4) rbacConnection {
-	conn := rbacConnection{}
-	conn.srcIp = binary.LittleEndian.AppendUint32(conn.srcIp, tupleV4.SrcAddr)
-	conn.dstIp = binary.LittleEndian.AppendUint32(conn.dstIp, tupleV4.DstAddr)
-	conn.dstPort = uint32(tupleV4.DstPort<<8 | tupleV4.DstPort>>8)
+func (r *Rbac) buildConnV4(buf *bytes.Buffer) (rbacConnection, error) {
+	var (
+		conn    rbacConnection
+		tupleV4 bpfSockTupleV4
+	)
+	if err := binary.Read(buf, binary.BigEndian, &tupleV4); err != nil {
+		log.Errorf("deserialize IPv4 FAILED, err: %v", err)
+		return conn, err
+	}
+	// srcIp and dstIp are big endian, and dstPort is little endian, which is consistent with authorization policy flushed to Kmesh
+	conn.srcIp = binary.BigEndian.AppendUint32(conn.srcIp, tupleV4.SrcAddr)
```

Comment:
we can use `netip.AddrFromSlice` if we make srcIp as type `netip.Addr`

Reply:
will be redesigned in the future if necessary